### PR TITLE
Darwin and Buildkite related improvements

### PR DESCRIPTION
**Pull request by johnalotoski**
* Modify darwin GC threshold to avoid double GC
* Reduce maxJobs in Darwin for lighter CPU, RAM and IO peak consumption
* Increase Darwin signer buildkite role for Catalina image
* Adds buildkite secret for plutus CI

* Note Darwin signer image currently requires interactive login to
  bootstrap apply.sh properly

### CODE REVIEW

**Comment by nrdxp** [comment]
> * Note Darwin signer image currently requires interactive login to
>   bootstrap apply.sh properly

Do you know what's causing this? It wasn't the case a few weeks ago?

File: roles/plutus.nix [hunk 1]
```diff
@@ -4,6 +4,14 @@
...
   imports = [ ../modules/buildkite-agent-containers.nix ];
 
+  deployment.keys = {
+    "buildkite-github-token" = {
+      keyFile = ../secrets/buildkite_github_token;
+      user = "buildkite-agent";
+    };
+  };
+
+
```
Comment (jonringer) [inline roles/plutus.nix:14]: nit: extra line

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

**Comment by jonringer** [comment]
> Note Darwin signer image currently requires interactive login to
bootstrap apply.sh properly

I know with xctools, you need to interactively "agree" to some ToS.

For compliance reasons, I could see why they would like for a human to have to manually approve something

**Comment by nrdxp** [comment]
> For compliance reasons, I could see why they would like for a human to have to manually approve something

I totally get that, but all one must do to get the startup script to provision the mac is login (and only on the signing machines). This was not happeneing before, so I just wonder what triggered the change.

**Review by jonringer — APPROVED** [review]
diff LGTM

**Pull request merged**